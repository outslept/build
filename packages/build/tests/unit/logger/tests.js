import { promises as fs } from 'fs'

import { normalizeOutput } from '@netlify/testing'
import tmp from 'tmp-promise'
import { test, expect } from 'vitest'

import { getSystemLogger } from '../../../lib/log/logger.js'

test('System logger writes to file descriptor', async () => {
  const { fd, cleanup, path } = await tmp.file()
  const mockProcess = {
    stdout: [],
  }
  const systemLog = getSystemLogger(mockProcess, false, fd)
  const error = new Error('Something went wrong')

  systemLog('Hello world', { object: true, problem: false }, error)

  const output = normalizeOutput(await fs.readFile(path, 'utf8'))

  expect(normalizeOutput(output)).toMatchSnapshot()
  expect(mockProcess.stdout.length).toBe(0)

  await cleanup()
})

test('System logger does not write to file descriptor when `debug: true`', async () => {
  const { fd, cleanup, path } = await tmp.file()
  const mockProcess = {
    stdout: [],
  }
  const systemLog = getSystemLogger(mockProcess, true, fd)
  const error = new Error('Something went wrong')

  systemLog('Hello world', { object: true, problem: false }, error)

  const output = normalizeOutput(mockProcess.stdout[0])

  expect(mockProcess.stdout.length).toBe(1)
  expect(normalizeOutput(output)).toMatchSnapshot()
  expect(await fs.readFile(path, 'utf8')).toBe('')

  await cleanup()
})
