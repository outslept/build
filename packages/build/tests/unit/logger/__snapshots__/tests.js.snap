// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`System logger does not write to file descriptor when \`debug: true\` 1`] = `
"Hello world {"object":true,"problem":false} Something went wrong Error: Something went wrong
STACK TRACE
STACK TRACE"
`;

exports[`System logger writes to file descriptor 1`] = `
"Hello world {"object":true,"problem":false} Something went wrong Error: Something went wrong
STACK TRACE
"
`;
